gnovm/pkg/gnolang/op_eval.go

```diff
@@ -56,6 +56,8 @@ func (m *Machine) doOpEval() {
 					_, ok = bi.SetString(x.Value[2:], 8)
 				case 'x', 'X':
 					_, ok = bi.SetString(x.Value[2:], 16)
+				case '0', '1', '2', '3', '4', '5', '6', '7':
+					_, ok = bi.SetString(x.Value, 8)
```

Comment:
nice :)

one note however: because of the above `if len(x.Value) > 2`, this will not catch octal literals like `07`. (It may still be parsed correctly by the below else block, but we need to reject literals like `09`). So please change it to make it work for literals like that, while still making sure to reject invalid literals like `0b` or `0x`.